[docs] Add docs for storage and tensors for quantized Tensor

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51817 [docs] Add docs for storage and tensors for quantized Tensor**

Differential Revision: [D26292464](https://our.internmc.facebook.com/intern/diff/D26292464)

jerryzh168 committed:
[docs] Add docs for storage and tensors for quantized Tensor

[ghstack-poisoned]

supriyar commented on docs/source/tensors.rst:
```diff
@@ -35,6 +39,8 @@ Boolean                    ``torch.bool``                                :class:
   significand bits. Useful when range is important, since it has the same
   number of exponent bits as ``float32``
+.. [3]
+  quantized 4-bit integer is stored as a 8-bit signed integer.
```
Can we also extend this comment to say that only embedding_bag kernels support this storage currently.

supriyar reviewed (approved):
Thanks for adding this!!

jerryzh168 committed:
Update on "[docs] Add docs for storage and tensors for quantized Tensor"




[ghstack-poisoned]

supriyar commented on docs/source/tensors.rst:
```diff
@@ -35,6 +39,8 @@ Boolean                    ``torch.bool``                                :class:
   significand bits. Useful when range is important, since it has the same
   number of exponent bits as ``float32``
+.. [3]
+  quantized 4-bit integer is stored as a 8-bit signed integer. Currently it's only supported in Embedding and EmbeddingBag operator.
```
4-bit is not supported for embedding, only embeddingBag :)

jerryzh168 committed:
Update on "[docs] Add docs for storage and tensors for quantized Tensor"


Differential Revision: [D26292464](https://our.internmc.facebook.com/intern/diff/D26292464)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[docs] Add docs for storage and tensors for quantized Tensor"


Differential Revision: [D26292464](https://our.internmc.facebook.com/intern/diff/D26292464)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[docs] Add docs for storage and tensors for quantized Tensor"


Differential Revision: [D26292464](https://our.internmc.facebook.com/intern/diff/D26292464)

[ghstack-poisoned]
